tests/compiler/bindings/esm.debug.d.ts

```diff
@@ -141,11 +141,11 @@ declare interface __Record8<TOmittable> {
   l: number | TOmittable;
   /** @type `f64` */
   m: number | TOmittable;
-  /** @type `~lib/string/String | null` */
+  /** @type `string | null` */
   n: string | null | TOmittable;
-  /** @type `~lib/typedarray/Uint8Array | null` */
+  /** @type `Uint8Array | null` */
   o: Uint8Array | null | TOmittable;
-  /** @type `~lib/array/Array<~lib/string/String> | null` */
+  /** @type `Array<~lib/string/String> | null` */
```

Comment:
@dcodeIO it looks like generic parameters somehow handles differently?

Reply:
This happens at

https://github.com/AssemblyScript/assemblyscript/blob/d5599823cc7af3b858967477a4c3520f5c7784b0/src/types.ts#L471-L475

where the `internalName` of a concrete class is constructed from the full names upon resolve around

https://github.com/AssemblyScript/assemblyscript/blob/d5599823cc7af3b858967477a4c3520f5c7784b0/src/resolver.ts#L2891-L2900

The requirement here is that internal names are guaranteed to be unique, which I guess would still be the case if the shorter names are used according to the convention that it applies to *global* elements (see below)?